Stripe/STPCard.m

```diff
@@ -169,44 +169,50 @@ - (NSString *)stripeObject {
     return @"card";
 }
 
-+ (NSArray *)requiredFields {
-    return @[@"id", @"last4", @"brand", @"exp_month", @"exp_year"];
-}
-
 + (nullable instancetype)decodedObjectFromAPIResponse:(nullable NSDictionary *)response {
-    NSDictionary *dict = [response stp_dictionaryByRemovingNullsValidatingRequiredFields:[self requiredFields]];
+    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
     if (!dict) {
         return nil;
     }
 
+    // required fields
+    NSString *stripeId = [dict stp_stringForKey:@"id"];
+    NSString *last4 = [dict stp_stringForKey:@"last4"];
+    NSString *rawBrand = [dict stp_stringForKey:@"brand"];
+    NSNumber *rawExpMonth = [dict stp_numberForKey:@"exp_month"];
+    NSNumber *rawExpYear = [dict stp_numberForKey:@"exp_year"];
+    if (!stripeId || !last4 || !rawBrand || !rawExpMonth || !rawExpYear) {
+        return nil;
+    }
+
     STPCard *card = [self new];
     card.address = [STPAddress new];
 
-    card.stripeID = dict[@"id"];
-    card.name = dict[@"name"];
-    card.last4 = dict[@"last4"];
-    card.dynamicLast4 = dict[@"dynamic_last4"];
-    card.brand = [self.class brandFromString:dict[@"brand"]];
+    card.stripeID = stripeId;
+    card.name = [dict stp_stringForKey:@"name"];
+    card.last4 = last4;
+    card.dynamicLast4 = [dict stp_stringForKey:@"dynamic_last4"];
+    card.brand = [self.class brandFromString:rawBrand];
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated"
     // This is only intended to be deprecated publicly.
     // When removed from public header, can remove these pragmas
     card.funding = [self.class fundingFromString:dict[@"funding"]];
```

Comment:
Is there a reason this isn't using `stp_stringForKey:`?

Reply:
oh nope, good catch!